### ISSUE TITLE
Staff Ignore Queue Paths

### ISSUE THREAD

**Issue by FrostyJTD** [OpenRCT2/OpenRCT2#1999]
I posted this over in the OpenRCT2 community forum, but I figured I'd throw it up here as well.

Staff have always entered queue lines, since the first game, and it's always been annoying. There's no reason for them to be in there. I'd love it if I could prevent them from doing so. Maybe as an option or a cheat?

**Comment by PFCKrutonium** [comment]
I could see this being an option, but it should definitely treat a queue connected to a ride different to a random piece of queue that isn't  - I've seen people use queue for paths before.

**Comment by FrostyJTD** [comment]
I thought about that too, but it's a simple check. An actual queue has the banner at the end of it, marking where the queue starts. Guests approach it before deciding if they want to go on a ride, so there's obviously something there. Just have staff not cross over that queue marker, and then queue path used as regular path will be treated as normal for staff.

**Comment by Patrik356b** [comment]
Maybe it's for janitors, if guests can litter/vomit in the ques. Never seen but that would explain it.

**Comment by PFCKrutonium** [comment]
I had a guest stuck in a puke loop in a queue, and while the guest was puking was the only time I have ever seen vomit in a queue.

**Comment by FrostyJTD** [comment]
guests don't litter nor puke in queues, under normal circumstances.

**Comment by IntelOrca** [comment]
Mechanics need to use queue lines to get to the entrance of a ride that has a station with only an entrance and not an exit.

**Comment by PFCKrutonium** [comment]
Could we detect that case and only allow them up the queue when it is true?

**Comment by duncanspumpkin** [comment]
The thing is entertainers in lines makes sense. Handymen in queues are occasionally required when using a cheat. Mechanics in queues are occasionally required. Guards are never required. I think it would just add a bunch of code for little improvement. I would prefer if we improved the handyman code so that they checked the queue for litter before entering it.

**Comment by IntelOrca** [comment]
Sure, although for the record I don't know if entertainers would enter the actual queue lines in real life. They also clip through all the peeps.

**Comment by FrostyJTD** [comment]
Like, yeah, entertainers at least kind of make sense in queue lines, but it does look very awkward and the question is do they actually affect the happiness of the people currently in the Queue? In all cases though, the reasons for staff being in queues are very fringe situations; cheating to get a peep to puke in the queue and the no exit station platform situation (which is something I didn't even know was possible but I'm 200% certain Orca knows more about RCT than I do). In both cases, dropping the staff inside of the queue would fix the problem. The only issue then would be if they consider exiting a queue the same as entering it in terms of their pathfinding logic.

**Comment by Gymnasiast** [comment]
Yes, they affect happiness, making them especially useful _in queues_ where the happiness of guests drops the longer they have to wait. (Simply opening guest windows where the entertainer is 'dancing' quickly shows that.) The clipping issue might be solved by making the staff walk off-centre - peeps stand in the centre and the queue is quite wide.

**Comment by Jimver** [comment]
I implemented this feature in #12140 however as some people said in the PR there are still some cases where guests can litter in the queues. So should this issue be closed?

**Comment by duncanspumpkin** [comment]
I'd like to eventually make it that staff check the queue line for litter prior to entering it but that will be done later as part of a bigger ai rework. Therefore i think this can stay open.
